applications/DEM_application/custom_elements/analytic_spheric_particle.cpp

```diff
@@ -55,37 +55,76 @@ int AnalyticSphericParticle::GetNumberOfCollisions()
     return mNumberOfCollidingSpheres;
 }
 
+int AnalyticSphericParticle::GetNumberOfCollisionsWithFaces()
+{
+    return mNumberOfCollidingSpheresWithFaces;
+}
+
+int AnalyticSphericParticle::GetNumberOfCollisionsWithEdges()
+{
+    return mNumberOfCollidingSpheresWithEdges;
+}
+
 void AnalyticSphericParticle::GetCollidingIds(array_1d<int, 4>& colliding_ids)
 {
     colliding_ids = mCollidingIds;
```

Comment:
I think this copy is unnecessary. It could be GetCollidingIds() with no arguments, and return mCollidingIds by reference.

Reply:
Actually, I think that passing an int by reference is overkill. Copying it is no more work than passing a pointer to it. Is'nt it?

Reply:
It's an array of 4 ints, actually. And outside the fuction you are now obliged to allocate a vector (and many others) when you could work with just references to the member variables, which are vectors as well.

Reply:
Oh, yeah, I misread it. Sorry! However, would a reference not expose the private variable? In any case, you should use noalias to speed it up.

Reply:
It is true that exposes the member variable. Maybe it should be accessed with a const method? 
Anyway, it was just a comment, if those allocations and copies can be avoided, great.